au4/src/projectscene/view/timeline/timelinecontext.cpp

```diff
@@ -205,6 +205,18 @@ double TimelineContext::positionToTime(double position, bool withSnap) const
     return result;
 }
 
+double TimelineContext::singleStepToTime(double position, Direction direction, bool snapEnabled) const
+{
+    double result = m_frameStartTime + position / m_zoom;
+    auto viewState = this->viewState();
+
+    if (viewState && snapEnabled) {
+        result = m_snapTimeFormatter->singleStep(result, viewState->snapType().val, viewState->isSnapTripletsEnabled().val, direction);
+    }
```

Comment:
It would be great if we could get values ​​with arbitrary snap settings (I feel like it would be both useful in the future and architecturally cleaner)
But it is not convenient to pass three arguments...
What do you think about combining this into one argument (structure) and also using this structure in viewState? 
Like: 

```
    struct Snap {
        SnapType type = SnapType::Bar;
        bool enabled = false;
        bool isSnapTriplets = false;
    };

    class IProjectViewState
    {
        void setIsSnapEnabled(bool enabled)
        {
            Snap s = this->snap();
            s.enabled = enabled;
            setSnap(s);
        }

        void setSnapType(SnapType type)
        {
            ...
        }
        void setIsSnapTripletsEnabled(bool enabled)
        {
            ...
        }
        virtual void setSnap(const Snap& s) const = 0;
        virtual Snap snap() const = 0;
        virtual muse::ValCh<Snap> snapChanged() const = 0;
    };


double TimelineContext::singleStepToTime(double position, Direction direction, const Snap& snap) const
{
...
}
```